snapshot-labs/subgrapher
# feat: Handle upgraded subgraphs similar to delegation subgraphs

## Summary

- Handle upgraded subgraphs similar to delegation subgraphs
- Add unit test cases for utils

## TODO:
- [ ] Add `NON_RESTRICTED_SUBGRAPH_KEY` to digital ocean


## how to test:
- Add a `NON_RESTRICTED_SUBGRAPH_KEY` in .env
- Run
```bash
curl --location 'http://localhost:3000/subgraph/arbitrum/5XqPmWe6gjyrJtFn9cLy237i4cWw2j9HcUJEXsP5qGtH' \
--header 'accept: application/json, multipart/mixed' \
--header 'accept-language: en-US,en;q=0.9' \
--header 'content-type: application/json' \
--data '{"query":"{\n  domains(first: 5) {\n    id\n    name\n    labelName\n    labelhash\n  }\n}","variables":{}}'
```
- Any other upgraded subgraph should work with this
